lib/probes/cassandra-driver.js

```diff
@@ -155,21 +154,87 @@ function serializeList (list) {
 }
 
 //
-// patch stream which is an event emitter and doesn't use callbacks.
+// patch stream in two ways
+// 1) bind the event emitter it returns to the async context.
+// 2) if a callback is supplied bind it to the async context.
 //
 function patchClientStream (client, consistencies) {
   if (typeof client.stream !== 'function') {
     logMissing('client.stream()')
     return
   }
-
   shimmer.wrap(client, 'stream', stream => function (...args) {
+    // if there is a callback bind to context
+    if (args.length && typeof args[args.length - 1] === 'function') {
+      args.push(ao.bind(args.pop()));
+    }
     // bind the emitter that stream returns
     return ao.bindEmitter(stream.apply(this, args))
   })
 }
 
+function patchClientExecute (client, consistencies) {
+  // 4.4.0+ _execute: async ƒ (query, params, execOptions)
+  if (typeof client._execute !== 'function' || client._execute.constructor.name !== 'AsyncFunction') {
+    logMissing('client._execute()')
+    return
+  }
+  shimmer.wrap(client, '_execute', execute => function (...args) {
+    const spanInfo = {
+      name: 'cassandra',
+      kvpairs: {
+        Spec: 'query',
+        Flavor: 'cql',
+      }
+    }
+
+    return ao.pInstrument(
+      // span info function
+      last => {
+        // Handle arguments
+        let [query, params, options] = normalizeArgs(...args)
+        if (!Array.isArray(params) && typeof params === 'object') {
+          options = args[1]
+          params = []
+        }
+        const consistency = options.consistency || {}
+        const keyspace = this.options.keyspace || ''
+
+        // Sanitize, if configured to do so
+        query = conf.sanitizeSql ? sanitize(query) : query
+
+        // Create a hash to store event k/v pairs
+        const kvpairs = spanInfo.kvpairs
+        // Keyspace is planned to be an alternative to Database.
+        // But Database is a required key, so add it too.
+        kvpairs.Keyspace = keyspace
+        kvpairs.Database = keyspace
+        kvpairs.ConsistencyLevel = consistencies[consistency] || 'one'
+        kvpairs.Query = query
+
+        // Serialize QueryArgs, if available
+        if (!conf.sanitizeSql && params.length) {
+          // Trim large values and ensure buffers are converted to strings
+          kvpairs.QueryArgs = serializeList(params)
+        }
+
+        // Truncate long queries
+        if (kvpairs.Query.length > 2048) {
+          kvpairs.Query = trim(2048)(kvpairs.Query)
+          kvpairs.QueryTruncated = true
+        }
+
+        return spanInfo
+      },
+      () => execute.apply(this, args),
+      conf
+    )
+  })
+}
+
+
 function patchClientInnerExecute (client, consistencies) {
+  // 4.4.0+ _execute: async ƒ (query, params, execOptions)
```

Comment:
not sure what this comment helps here, should it be `_innerExecute`

Reply:
nice catch. this was a working note documenting the new signature that i originally made in `patchClientInnerExecute()`. that was copied when i cut/pasted/modified `patchClientInnerExecute()` to create `patchClientExecute()` so it then appeared in both places. i've removed both comments and added a more meaningful comment where the decision on which function to patch is made.